### PR TITLE
fix: reduce use of `length/1` whenever not needed

### PR DESCRIPTION
**Pull request by hauleth**

### CODE REVIEW

File: lib/logflare/logs/processor.ex [hunk 1]
```diff
@@ -27,17 +27,35 @@ defmodule Logflare.Logs.Processor do
...
+      :telemetry.execute(
+        [:logflare, :logs, :processor, :ingest, :logs],
+        %{
+          count: length(batch)
+        },
+        metadata
+      )
 
-      {result, Map.merge(metadata, %{success: result == :ok})}
+      :telemetry.span([:logflare, :logs, :processor, :ingest, :ingest], metadata, fn ->
```
Comment (Ziinc) [inline lib/logflare/logs/processor.ex:43]:
```suggestion
      :telemetry.span([:logflare, :logs, :processor, :ingest, :store], metadata, fn ->
```

File: lib/logflare/sources.ex [hunk 2]
```diff
@@ -341,15 +341,13 @@ defmodule Logflare.Sources do
     %{source | metrics: new_metrics}
   end
 
-  def valid_source_token_param?(string) when is_binary(string) do
-    case String.length(string) === 36 && Ecto.UUID.cast(string) do
+  def valid_source_token_param?(string) do
```
Comment (Ziinc) [inline lib/logflare/sources.ex:344]: this uuid check should use this Ecto.UUID.dump instead, see here
https://github.com/Logflare/logflare/pull/1617

File: lib/logflare/utils/list.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
+defmodule Logflare.Utils.List do
```
Comment (Ziinc) [inline lib/logflare/utils/list.ex:1]: doctests missing

**Review by Ziinc — CHANGES_REQUESTED** [review]

**Review by Ziinc — APPROVED** [review]

**Comment by Ziinc** [comment]
@hauleth could you run `mix format` 🙏

**Comment by hauleth** [comment]
Shure, in a moment.

**Pull request merged**